pandas/core/indexes/base.py

```diff
@@ -4259,7 +4266,8 @@ def _concat_same_dtype(self, to_concat, name):
         Concatenate to_concat which has the same class.
         """
         # must be overridden in specific classes
-        klasses = (ABCDatetimeIndex, ABCTimedeltaIndex, ABCPeriodIndex, ExtensionArray)
+        klasses = (ABCDatetimeIndex, ABCTimedeltaIndex, ABCPeriodIndex, ExtensionArray,
+                   ABCIntervalIndex)
```

Comment:
Without this, if the first element is a IntervalIndex, a"side" parameter is passed to ``_shallow_copy_with_infer`` ind the call will fail.